homeassistant/components/plugwise/__init__.py

```diff
@@ -21,7 +22,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: PlugwiseConfigEntry) ->
 
     coordinator = PlugwiseDataUpdateCoordinator(hass)
     await coordinator.async_config_entry_first_refresh()
-    migrate_sensor_entities(hass, coordinator)
+    await async_migrate_sensor_entities(hass, coordinator)
+    await async_migrate_plugwise_entry(hass, coordinator, entry)
```

Comment:
Tried that, can't make it work. I need to access the coordinator to be able to pick up the firmware version.

Reply:
Ooh right. Hmm. Would it be possible to create a new connection during migration?

Reply:
Unless... I store it as part of the `config-entry`, would that work?

Reply:
I will try to get the required version from the `device_registry.sw_version`, that should be the simplest method.

Reply:
But wait (this is the moment I actually realize what we are doing here). Why don't we fetch this on runtime? Because you can change the update interval of a coordinator on runtime, why do we need to store the version in the entry?

Reply:
Ok, the PR has been reworked.